src/HTMLTokenizer/HTMLTokenizer.cc

```diff
@@ -0,0 +1,281 @@
+#include <node.h>
+#include <string>
+#include <nan.h>
+#include <stdio.h>
+#include <iostream>
+
+#include "HTMLTokenizer.h"
+
+using namespace v8;
+
+enum STATES {
+	ILLEGAL = -1,
+	INITIAL,
+	CONTENT,
+	COMPONENT,
+	COMMENT,
+	END
+};
+
+std::string v8_to_std_string(v8::Local<v8::Value> v8Val) {
+    String::Utf8Value UtfValue(v8Val->ToString());
+    std::string str = std::string(*UtfValue);
+    
+    return str;
+}
+
+Persistent<Function> HTMLTokenizer::constructor;
+
+HTMLTokenizer::~HTMLTokenizer() {
+}
+
+HTMLTokenizer::HTMLTokenizer() {
+}
+
+void HTMLTokenizer::Init(Handle<Object> exports, Handle<Object> module) {
+    NanScope();
+
+    Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+    tpl->SetClassName(NanNew("HTMLTokenizer"));
+    tpl->InstanceTemplate()->SetInternalFieldCount(3);
+
+    NODE_SET_PROTOTYPE_METHOD(tpl, "setHTMLString", setHTMLString);
+    NODE_SET_PROTOTYPE_METHOD(tpl, "next", next);
+    // NODE_SET_PROTOTYPE_METHOD(tpl, "content", content);
+    // NODE_SET_PROTOTYPE_METHOD(tpl, "component", component);
+    // NODE_SET_PROTOTYPE_METHOD(tpl, "comment", comment);
+    // NODE_SET_PROTOTYPE_METHOD(tpl, "initial", initial);
```

Comment:
just decided to make it hidden from JS, because otherwise it couldn't be simply called from C++ code -- you have to pass specific args from V8 constructor conventions and other staff...